Modules/_asynciomodule.c

```diff
@@ -3724,6 +3724,9 @@ module_clear(PyObject *mod)
 
     Py_CLEAR(state->context_kwname);
 
+    _PyThreadStateImpl *ts = (_PyThreadStateImpl *)_PyThreadState_GET();
+    Py_CLEAR(ts->asyncio_running_loop);
```

Comment:
This only clears the `asyncio_running_loop` in one thread (probably the main thread during shutdown). But each thread has its own `asyncio_running_loop` field.

Reply:
The general case is that there's only one event loop running in main thread, but you are correct this won't work in case of multiple loops. We would probably have to traverse the thread states and clear it for all of them